src/cplus_plugin/api/scenario_task_output_download.py

```diff
@@ -0,0 +1,70 @@
+import concurrent.futures
+import datetime
+import json
+import os
+import traceback
+import typing
+from zipfile import ZipFile
+
+import requests
+from qgis.core import Qgis
+from .multipart_upload import upload_part
+from .request import (
+    CplusApiRequest,
+    JOB_COMPLETED_STATUS,
+    JOB_STOPPED_STATUS,
+    CHUNK_SIZE,
+)
+from .scenario_task_api_client import ScenarioAnalysisTaskApiClient
+from ..conf import settings_manager, Settings
+from ..models.base import Activity, NcsPathway
+from ..models.base import ScenarioResult
+from ..tasks import ScenarioAnalysisTask
+from ..utils import FileUtils, CustomJsonEncoder, todict
+
+
+class ScenarioTaskOutputDownload(ScenarioAnalysisTaskApiClient):
```

Comment:
@zamuzakki can you add doc strings to the class and public methods? there are several changes, I added a comment here but please add to other places.